jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/BuildDockerTask.java

```diff
@@ -87,9 +75,7 @@ public void dockerClient(Action<? super DockerClientParameters> action) {
   }
 
   @TaskAction
-  public void buildDocker()
-      throws IOException, BuildStepsExecutionException, CacheDirectoryCreationException,
-          MainClassInferenceException {
+  public void buildDocker() {
```

Comment:
There are pros and cons, but to me, the downside of hiding each exception under the uber `Exception` seems to weigh more than the advantages. I would go with `buildDocker() throws IOException, BuildStepsExecutionException, ...` and have `handleJibException(<some function type> buildTask)` like

```java
  try {
    buildTask.run();
  } catch (InvalidAppRootException ex) {
    ...
```

so that whenever we introduce a new type of exception, we don't forget to handle it properly. Implementation-wise, passing a callable function is one way like above, but having an abstract base class could be another. But either of them will introduce some more code and that's the downside.

Reply:
Yeah, good point. I think I'll close this and we can figure out a better long-term solution in the future.